File: crates/klyron_bundler/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_bundler {
BundlerBuilder::BundlerBuilder() : config_() {}
BundlerBuilder& BundlerBuilder::enabled(bool v) { config_.enabled = v; return *this; }
BundlerBuilder& BundlerBuilder::verbose(bool v) { config_.verbose = v; return *this; }
BundlerConfig BundlerBuilder::build() { return config_; }
}

File: crates/klyron_bundler/bindings/rust/builder.rs
use crate::config::BundlerConfig;

pub struct BundlerBuilder {
    config: BundlerConfig,
}

impl BundlerBuilder {
    pub fn new() -> Self { Self { config: BundlerConfig::default() } }
    pub fn build(self) -> BundlerConfig { self.config }
}

impl Default for BundlerBuilder {
    fn default() -> Self { Self::new() }
}
